modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc

```diff
@@ -20,45 +20,156 @@ To provision bare-metal hosts to the cluster by using the BMO, you must set the
-.. Save the following YAML in the `bmh.yaml` file:
+. Create a configuration file for the bare-metal node. Depending on whether you are using a static configuration or a DHCP server, use one of the following example `bmh.yaml` files, replacing values in the YAML to match your environment:
++
+.Static configuration `bmh.yaml`
 +
 [source,yaml]
 ----
 ---
 apiVersion: v1
 kind: Secret
 metadata:
-  name: worker1-bmc
+  name: openshift-worker-<num>-network-config-secret # <1>
+  namespace: openshift-machine-api
+type: Opaque
+stringData:
+ nmstate: | # <2>
+  interfaces: # <3>
+  - name: <nic1_name> # <4>
+    type: ethernet
+    state: up
+    ipv4:
+      address:
+      - ip: <ip_address> # <5>
+        prefix-length: 24
+      enabled: true
+  dns-resolver:
+    config:
+      server:
+      - <dns_ip_address> # <6>
+  routes:
+    config:
+    - destination: 0.0.0.0/0
+      next-hop-address: <next_hop_ip_address> # <7>
+      next-hop-interface: <next_hop_nic1_name> # <8>
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: openshift-worker-<num>-bmc-secret # <1>
   namespace: openshift-machine-api
 type: Opaque
 data:
-  username: <base64_of_uid>
-  password: <base64_of_pwd>
+  username: <base64_of_uid> # <9>
+  password: <base64_of_pwd> # <9>
 ---
 apiVersion: metal3.io/v1alpha1
 kind: BareMetalHost
 metadata:
-  name: worker1
+  name: <openshift-worker-<num> # <1>
   namespace: openshift-machine-api
 spec:
+  online: true
+  bootMACAddress: <nic1_mac_address> # <10>
   bmc:
-    address: <protocol>://<bmc_url> <1>
-    credentialsName: "worker1-bmc"
-  bootMACAddress: <nic1_mac_address>
-  externallyProvisioned: false <2>
+    address: <protocol>://<bmc_url> # <11>
+    credentialsName: openshift-worker-<num>-bmc-secret # <1>
+    disableCertificateVerification: false
   customDeploy:
     method: install_coreos
+  userData:
+    name: worker-user-data-managed
+    namespace: openshift-machine-api
+  rootDeviceHints:
+    deviceName: <root_device_hint> # <12>
+  preprovisioningNetworkDataName: openshift-worker-<num>-network-config-secret # <1>
+----
++
+--
+<1> Replace `<num>` with a unique worker number for the bare-metal nodes in the `name`, `credentialsName`, and `preprovisioningNetworkDataName` fields.
+<2> Add the NMState YAML syntax to configure the host interfaces. To configure the network interface for a newly created node, specify the name of the secret that has the network configuration. Follow the `nmstate` syntax to define the network configuration for your node. See "Preparing the bare metal node" for details on configuring NMState syntax.
+<3> Optional: If you have configured the network interface with `nmstate`, and you want to disable an interface, set `state: up` with the IP addresses set to `enabled: false` as shown in the following example:
++
+[source,yaml]
+----
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: openshift-worker-<num>-network-config-secret
+  namespace: openshift-machine-api
+ # ...
+interfaces:
```

Comment:
This `interfaces` is not indented that same level as the `interfaces` stanza in the main YAML block. Is this showing an alternate configuration of the whole first section? 

* **IF** the `# ...` on line 102 is hiding structure, it needs to be fully written out to get to `interfaces`
* **IF** the indent is right and line 102 is _not_ hiding structure, consider making this a separate config file example. Putting this much in a callout is hard to follow.
* in any case, I would suggest making these different config examples substeps of step one, i.e.

1. Create a configuration file for the bare-metal node.
  * If you are using a static configuration, use the following example `bmh.yaml` file, replacing values in the YAML to match your environment:
  [YAML block for static]
  * (Possible new bullet for the `state: up` / `enabled: false` flavor?)
  [YAML block for it]
  * If you are using a DHCP server, use the following example `bmh.yaml` file, replacing values in the YAML to match your environment:
  [YAML block for DHCP]

I am not sure how well I have explained what I am trying to say via GH comment, I would be happy to chat with you realtime if that would be more helpful :sweat_smile: